test/replication.rest.test.js

```diff
@@ -0,0 +1,484 @@
+    describe('Car model', function() {
+      it('rejects anonymous READ', function(done) {
+        listCars().expect(401, done);
+      });
+
+      it('rejects anonymous WRITE', function(done) {
+        createCar().expect(401, done);
+      });
+
+      it('allows EMERY to READ', function(done) {
+        listCars()
+          .set('Authorization', emeryToken)
+          .expect(200, done);
+      });
+
+      it('denies EMERY to WRITE', function(done) {
+        createCar()
+          .set('Authorization', emeryToken)
+          .expect(401, done);
+      });
+
+      it('allows ALICE to READ', function(done) {
+        listCars()
+          .set('Authorization', aliceToken)
+          .expect(200, done);
+      });
+
+      it('denies ALICE to WRITE', function(done) {
+        createCar()
+          .set('Authorization', aliceToken)
+          .expect(401, done);
+      });
+
+      it('allows PETER to READ', function(done) {
+        listCars()
+          .set('Authorization', peterToken)
+          .expect(200, done);
+      });
+
+      it('allows PETER to WRITE', function(done) {
+        createCar()
+        .set('Authorization', peterToken)
+        .expect(200, done);
+      });
+
+      function listCars() {
+        return request.get('/Cars');
+      }
+
+      function createCar() {
+        return request.post('/Cars').send({ model: 'a-model' });
+      }
+    });
+  });
+
+  describe('sync with model-level permissions', function() {
+    describe('as anonymous user', function() {
+      it('rejects pull from server', function(done) {
+        RemoteCar.replicate(LocalCar, expectHttpError(401, done));
+      });
+
+      it('rejects push to the server', function(done) {
+        LocalCar.replicate(RemoteCar, expectHttpError(401, done));
+      });
+    });
+
+    describe('as user with READ-only permissions', function() {
+      beforeEach(function() {
+        setAccessToken(emeryToken);
+      });
+
+      it('rejects pull from server', function(done) {
+        RemoteCar.replicate(LocalCar, expectHttpError(401, done));
+      });
+
+      it('rejects push to the server', function(done) {
+        LocalCar.replicate(RemoteCar, expectHttpError(401, done));
+      });
+    });
+
+    describe('as user with REPLICATE-only permissions', function() {
+      beforeEach(function() {
+        setAccessToken(aliceToken);
+      });
+
+      it('allows pull from server', function(done) {
+        RemoteCar.replicate(LocalCar, function(err, conflicts, cps) {
+          if (err) return done(err);
+          if (conflicts.length) return done(conflictError(conflicts));
+
+          LocalCar.find(function(err, list) {
+            if (err) return done(err);
+            expect(list.map(carToString)).to.include.members(serverCars);
+            done();
+          });
+        });
+      });
+
+      it('rejects push to the server', function(done) {
+        LocalCar.replicate(RemoteCar, expectHttpError(401, done));
```

Comment:
At the moment, replicate = pull changes. I am fine with renaming it if it makes the intent more clear. It's much better than relying on docs, as most people don't read them. Should I use `PULLCHANGES` instead of `REPLICATE`?